google/resumable_media/_helpers.py

```diff
@@ -369,6 +371,19 @@ def add_query_parameters(media_url, query_params):
     return urlunsplit((scheme, netloc, path, query, frag))
 
 
+def _is_decompressive_transcoding(response, get_headers):
+    """Returns True if the object was served decompressed and the "x-goog-stored-content-encoding"
+    header is "gzip". See more at: https://cloud.google.com/storage/docs/transcoding#transcoding_and_gzip
+    Args:
+        response (~requests.Response): The HTTP response object.
+        get_headers (callable: response->dict): returns response headers.
+    Returns:
+        bool: Returns True if the "x-goog-stored-content-encoding" header is "gzip"; otherwise, False.
+    """
+    headers = get_headers(response)
+    return headers.get(_STORED_CONTENT_ENCODING_HEADER) == "gzip"
```

Comment:
You also want to check if `content-encoding` was not in the response headers with the value `gzip`. There's an edge case in that GCS return compressed data with `content-encoding: gzip`.

Reply:
That's right. Thanks Frank for catching that. I've added the check and a test alongside.